### PR TITLE
boards: arm: Add support for EVK-NINA-B3

### PR DESCRIPTION
**Pull request by ghost**
Add support for u-blox EVK-NINA-B3 which uses the nRF52840.

This board is similar to the nRF52840dk_nrf52840 with
different pin assignments on the header pins and not having
the debug-in and shield SWD headers.

Tested with blinky, button, and Bluetooth peripheral_hr

Signed-off-by: Bob Recny <bob.recny@u-blox.com>

### CODE REVIEW

**Review by carlescufi — APPROVED** [review]
LGTM, thank you!

**Comment by brec-u-blox** [comment]
Just checking status. Thanks.

File: boards/arm/ubx_evkninab3_nrf52840/doc/index.rst [hunk 1]
```diff
@@ -0,0 +1,323 @@
...
+Power (J1)
+
++-------+--------------+-------------------------+
+| PIN # | Signal Name  | NINA-B30x Functions     |
++=======+==============+=========================+
+| 1     | N/C          | N/A                     |
++-------+--------------+-------------------------+
+| 2     | VDD_IO       | N/A                     |
++-------+--------------+-------------------------+
+| 3     | RESET        | P0.21 / RESET           |
```
Comment (anangl) [inline boards/arm/ubx_evkninab3_nrf52840/doc/index.rst:130]: P0.18

File: boards/arm/ubx_evkninab3_nrf52840/ubx_evkninab3_nrf52840.dts [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+		pwm_led0: pwm_led_0 {
+			pwms = <&pwm0 13>;
+		};
+		pwm_led1: pwm_led_1 {
+			pwms = <&pwm0 32>;
+		};
+		pwm_led2: pwm_led_2 {
+			pwms = <&pwm0 25>;
+		};
+		red_pwm_led: red_pwm_led {
```
Comment (anangl) [inline boards/arm/ubx_evkninab3_nrf52840/ubx_evkninab3_nrf52840.dts:52]: Why do you define these LEDs twice? Wouldn't it be sufficient to use only aliases, e.g.:
```
red-pwm-led = &pwm_led_0;
```

File: boards/arm/ubx_evkninab3_nrf52840/doc/index.rst (hunk 1 above)
Comment (brec-u-blox) [reply boards/arm/ubx_evkninab3_nrf52840/doc/index.rst]: Fixed, thanks

File: boards/arm/ubx_evkninab3_nrf52840/ubx_evkninab3_nrf52840.dts (hunk 2 above)
Comment (brec-u-blox) [reply boards/arm/ubx_evkninab3_nrf52840/ubx_evkninab3_nrf52840.dts:43]: Done - thanks for the suggestion.

**Review by anangl — APPROVED** [review]

**Pull request merged**